packages/ui-components/src/icon/jlicon.tsx

```diff
@@ -333,26 +460,128 @@ export class JLIcon {
     return component;
   }
 
+  protected _initSvg({
+    title,
+    uuid
+  }: { title?: string; uuid?: string } = {}): HTMLElement | null {
+    const svgElement = JLIcon.resolveSvg(this);
+
+    if (!svgElement) {
+      // bail on null svg element
+      return svgElement;
+    }
+
+    if (svgElement.tagName !== 'parsererror') {
+      // svgElement is an actual svg node, augment it
+      svgElement.dataset.icon = this.name;
+
+      if (uuid) {
+        svgElement.dataset.iconId = uuid;
+      }
+
+      if (title) {
+        Private.setTitleSvg(svgElement, title);
+      }
+    }
+
+    return svgElement;
+  }
+
   readonly name: string;
+
+  /**
+   * A React component that will create the icon.
+   *
+   * @param className - a string that will be used as the class
+   * of the container element. Overrides any existing class
+   *
+   * @param container - a preexisting DOM element that
+   * will be used as the container for the svg element
+   *
+   * @param label - text that will be displayed adjacent
+   * to the icon
+   *
+   * @param title - a tooltip for the icon
+   *
+   * @param tag - if container is not explicitly
+   * provided, this tag will be used when creating the container
+   *
+   * @param propsStyle - style parameters that get passed to TypeStyle in
+   * order to generate a style class. The style class will be added
+   * to the icon container's classes, while the style itself will be
+   * applied to any svg elements within the container.
+   */
   readonly react: JLIcon.IReact;
 
+  readonly render: (container: HTMLElement, props?: JLIcon.IProps) => void;
+  readonly unrender: (container: HTMLElement) => void;
+
   protected _className: string;
+  protected _loading: boolean;
+  protected _svgReplaced = new Signal<this, void>(this);
   protected _svgstr: string;
   protected _uuid: string;
+
+  // needed due to the quirks of the current implementation of IRenderer
+  protected _icon = this;
+  protected _rendererOptions = {};
 }
 
 /**
  * A namespace for JLIcon statics.
  */
 export namespace JLIcon {
+  /**************
+   * interfaces *
+   **************/
+
   /**
-   * The type of the JLIcon contructor params
+   * The simplest possible interface for defining a generic icon.
    */
-  export interface IOptions {
-    name: string;
+  export interface IIcon {
```

Comment:
Do we need this interface since we already have one in rendermine? Can we import that one here?

Reply:
Originally I was just following the same convention as for IFileType: the "real" interface lives in docregistry, while rendermime-interfaces contains a "seperate but equal" reimplmentation. On reflection, "because IFileType did it" doesn't make for the strongest design argument. @jasongrout What's your opinion on how interface separation should work here?

Reply:
I do kind of like having the two separate `IIcon` interfaces:

- rendermime-interfaces obviously can't depend on ui-components, but it doesn't really make that much sense for ui-components to depend on rendermime-interfaces

- This way, I'm free to make tweaks to the "real" `IIcon` while the rendermime-interfaces `IIcon` remains forever frozen in amber (of course, I could acheive more or less the same thing by importing `IIcon` from rendermime and then extending it, but for such a simple interface I think it's kind of a wash).

Reply:
I think in general we should prefer to inherit from `rendermime-interfaces` to keep us honest.